lib/plugins/Tests/Tests.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const path = require('path');
+const fs = require('fs');
+
+class Tests {
+  constructor() {
+    this.commands = {
+      test: {
+        commands: {
+          integration: {
+            usage: 'Command for integration testing.',
+            lifeCycleEvents: [
+              'createFile',
+            ],
+          },
+        },
+      },
+    };
+
+    this.hooks = {
+      'test:integration:createFile': this.createFile,
+    };
+  }
+
+  createFile() {
```

Comment:
That's a good idea. Do you know a possibility to read from the console? That was the problem I was facing during the implementation.

Seems like NPM tests are also using files to test their CLI (https://github.com/npm/npm/blob/2a5977e0c65b244e92d848fcd56f2f80ba8cdf3b/test/tap/outdated-depth.js).